### PR TITLE
Refactor project structure, update OTel version to 0.99.0

### LINKED ISSUE #109: Update githubactionsreceiver
Our "fork" of the githubactions receiver (w/ support for logs) is lagging a bit behind the one proposed upstream (https://github.com/krzko/opentelemetry-collector-contrib/tree/feat-add-githubactionseventreceiver/receiver) which includes a few fixes. we should rebase https://github.com/grafana/opentelemetry-collector-contrib/tree/feat-add-githubactionseventreceiver

### LINKED ISSUE #113: Refactor project structure
The current setup of the project is making it a bit hard to keep it up to date with the upstream collector. We are currently blocked because trying to update to otel >=0.99 (which is needed to keep the `githubactionsreceiver` up to date with upstream) breaks `mdatagen`. 

We should restructure the repo to be more in line with the [contrib repo](https://github.com/open-telemetry/opentelemetry-collector-contrib)

### PR DESCRIPTION
**Pull request by Elfo404**
Refactors project structure to be more inline with what is done in the [contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib) repo. Largely inspired by [liatrio-otel-collector](https://github.com/liatrio/liatrio-otel-collector). This is to use updated tooling that wouldn't otherwise work (ie. mdatagen & ocb that were moved from the contrib repo to core and don't work anymore with bingo) and make sure we can move forward by emulating what the contrib repo does instead of reinventing the wheel every time something breaks due to upstream tooling updates.

On top of that:
- Updates the GHA receiver to https://github.com/grafana/opentelemetry-collector-contrib/tree/feat-add-githubactionseventreceiver-2 - which is ~in sync with upstream (with added support for logs)
- Updates otel version to 0.99
- Refactors/fixes Drone config due to project structure changes
- removes bingo

Fixes #113 
Fixes #109 

tips for reviewing:
- 98b6d8e2154b0f02a316a5ce1301dcc437029259 is the actual project structure change:
  - in https://github.com/grafana/grafana-ci-otel-collector/pull/114/commits/98b6d8e2154b0f02a316a5ce1301dcc437029259#diff-39f79eeaa4ceb3340429dabf0855febb0414deb9790f0d676ebf309e9915edf3 the builder config file has been renamed and includes the updates to otel versions for each component
  - everything in `receiver/dronereceiver`, `internal/semvonv` and `internal/traceutils` is just moved from the old directories, with a few modifications due to deprecated methods & updated tooling
  - https://github.com/grafana/grafana-ci-otel-collector/pull/114/commits/98b6d8e2154b0f02a316a5ce1301dcc437029259#diff-e607ec0aa5cc1bb54b418caa5ae0f5d10116ed19a4d016748dff76b68a11f8d4 (and the module itself) now hosts all the tooling
- e98907e4c419c79eba77adcb22230c068f8b9b0a removes `multimod` and `genqlient` tools as unused
- https://github.com/grafana/grafana-ci-otel-collector/pull/114/commits/3dae01a5685b731366a3c7401e5f53ce7e1cb9b2 fixes a linter issue, https://github.com/grafana/grafana-ci-otel-collector/pull/114/commits/f0fc8526a3921bd5abe8ac2ff90768cb92074bde removes porto, govulncheck and goreleaser as they were not used (most of the changes are autogenerated)
- a8f3329a2030e0afe9c0e1a33b9507bdc86bed06 this fixes CI and updates the docker image being built to be way smaller than it used to (34.2MB vs 2.8GB)

### CODE REVIEW

File: golangci.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+---
+run:
+  go: '1.21'
```
Comment (zerok) [inline golangci.yaml:3]:
```suggestion
  go: '1.22'
```

Just so that it is consistent with the `go.mod` file 🙂

File: Makefile.Common [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+CROSSLINK := $(TOOLS_BIN_DIR)/crosslink
+OSV := $(TOOLS_BIN_DIR)/osv-scanner
+
+.PHONY: lint
+lint:
+	$(LINT) run
+
+.PHONY: tidy
+tidy:
+	-rm go.sum
```
Comment (dsotirakis) [inline Makefile.Common:53]: need the dash here?

**Review by dsotirakis — APPROVED** [review]
🚀

File: Makefile.Common (hunk 2 above)
Comment (dsotirakis) [reply Makefile.Common:53]: okay good then!

**Pull request merged**